### DISCUSSION TITLE
has_many relation result order issue.

### DISCUSSION THREAD

**Discussion by zw963** [luckyframework/lucky#1718] [Q&A] [answered]
Following is a example.

```crystal
abstract class BaseModel < Avram::Model
  # ...

  def initialize
    defaults &.id.desc_order
  end
end

class Post < BaseModel
  table do
    # columns ...

    has_many comments : Comment
  end
end

class Comment < BaseModel
  table do
    # columns ...

    belongs_to post : Post
  end
end
```

When i try get all comments of a post, what i expected is all comments can order by id desc.

```crystal
post = PostQuery.find(env.params.url["post_id"])

post.comments
```

But, when i check log, there is no `order by` in it.

2022-08-11T12:20:43.919739Z   INFO - avram.query:  -- local: {"model" => "Comment", "query" => "SELECT * FROM comments WHERE comments.post_id = $1", "args" => "[\"1\"]", "duration" => "1.22ms"}

I try following [querying-records#queries-with-defaults](https://luckyframework.org/guides/database/querying-records#queries-with-defaults), query like this:

```crystal
ReportQuery.new.post_id(post.id)
```

It not work, so, i have to add `id.desc_order` to get it work.

```crystal
ReportQuery.new.post_id(post.id).id.desc_order
```

Following is my question:

~~1. Why the `defaults &.id.desc_order` in base_model.cr not work?~~
2. has_many relation like `post.comments` always return a Array, can i return a Relation instead? or, can i specify the order when `has_many comments : Comment`?

Thanks

**EDIT**

I guess i add code to wrong place ....

**Comment by jwoertink** [comment, marked as answer]
Yeah, it looks like you added query code to the model. That `initialize` was never being called. 

As for doing the relation, that's not possible at the moment. https://github.com/luckyframework/avram/issues/85 

I want to add it in, but I tried, and it's really hard. https://github.com/luckyframework/avram/pull/584

For now, what I do is just create a new method on the model that builds the query how I need.

```crystal
class Post < BaseModel

  def blocked_comments : CommentQuery
    CommentQuery.new.post_id(self.id).blocked(true)
  end
end
```
